### PR TITLE
Fmt

### PR DESCRIPTION
**Pull request by xcoldhandsx**

### CODE REVIEW

File: projects/fmt/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+#
+################################################################################
+
+FROM sweetvishnya/ubuntu20.04-sydr-fuzz
+
+MAINTAINER Andrey Fedotov
+
+# Install build dependencies.
+RUN apt-get update && apt-get install -y ninja-build
```
Comment (SweetVishnya) [inline projects/fmt/Dockerfile:23]: ninja is already in base container

File: projects/fmt/build.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+-DCMAKE_BUILD_TYPE=Debug \
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=Off \
+-DFMT_FUZZ_LDFLAGS="$(find /usr/lib/clang -name libclang_rt.fuzzer-x86_64.a)"
+
+cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:48]: parallel build

File: projects/fmt/build.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+cp -r bin ./fuzzers
+
+# Sydr
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:52]: clean build dir?

File: projects/fmt/build.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+-DCMAKE_EXE_LINKER_FLAGS="-lpthread" \
+-DCMAKE_BUILD_TYPE=Debug \
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:66]: build in parallel

File: projects/fmt/build.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+cp -r bin ./sydr
+
+# Coverage
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:70]: clean build

File: projects/fmt/build.sh [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+-DCMAKE_EXE_LINKER_FLAGS="-lpthread -fprofile-instr-generate" \
+-DCMAKE_BUILD_TYPE=Debug \
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:84]: build in parallel

File: projects/fmt/build.sh [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+#
+################################################################################
+
+#create zip files with initial corpus, taken from version control.
+#for f in $(ls fuzzers/initial_corpus/) ;do
+#  zip -j -r $OUT/fuzzer_${f}_seed_corpus.zip fuzzers/initial_corpus/$f
+#done
+
+mkdir build
+mkdir fuzzers
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:25]: no need to mkdir fuzzers sydr cov

File: projects/fmt/build.sh [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=Off \
+-DFMT_FUZZ_LDFLAGS="$(find /usr/lib/clang -name libclang_rt.fuzzer-x86_64.a)"
+
+cmake --build .
+
+cp -r bin ./fuzzers
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:50]: cp -r bin ../fuzzers

File: projects/fmt/build.sh [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+cmake --build .
+
+cp -r bin ./sydr
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:68]: cp -r bin ../sydr

File: projects/fmt/build.sh [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+cmake --build .
+
+cp -r bin ./cov
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:86]: cp -r bin ../cov

File: projects/fmt/chrono_duration.toml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+exit-on-time = 7200
+
+[sydr]
+args = "-l debug -s 90 --app-timeout 180 -m 8192"
+target = "/fmt/build/sydr/fuzzer_chrono_duration @@"
```
Comment (SweetVishnya) [inline projects/fmt/chrono_duration.toml:20]: /fmt/sydr/fuzzer_chrono_duration

remove build in other configs?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/fmt/build.sh (hunk 5 above)
Comment (xcoldhandsx) [reply projects/fmt/build.sh:70]: Not necessary, rebuild works fine.

File: projects/fmt/build.sh [hunk 12]
```diff
@@ -45,9 +42,11 @@ cmake .. \
 -DFMT_FUZZ_LINKMAIN=Off \
 -DFMT_FUZZ_LDFLAGS="$(find /usr/lib/clang -name libclang_rt.fuzzer-x86_64.a)"
 
-cmake --build .
+CMAKE_BUILD_PARALLEL_LEVEL=6 cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:45]: $(nproc)

File: projects/fmt/build.sh [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+-DCMAKE_EXE_LINKER_FLAGS="-lpthread" \
+-DCMAKE_BUILD_TYPE=Debug \
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+CMAKE_BUILD_PARALLEL_LEVEL=6 cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:65]: $(nproc)

File: projects/fmt/build.sh [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+-DCMAKE_EXE_LINKER_FLAGS="-lpthread -fprofile-instr-generate" \
+-DCMAKE_BUILD_TYPE=Debug \
+-DCMAKE_CXX_STANDARD=14 \
+-DFMT_DOC=Off \
+-DFMT_TEST=Off \
+-DFMT_SAFE_DURATION_CAST=On \
+-DFMT_FUZZ=On \
+-DFMT_FUZZ_LINKMAIN=ON
+
+CMAKE_BUILD_PARALLEL_LEVEL=6 cmake --build .
```
Comment (SweetVishnya) [inline projects/fmt/build.sh:85]: $(nproc)

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Pull request merged**